remove-webhook/remove-webhook.sh

```diff
@@ -1,24 +1,81 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-if [ -n "$DEBUG" ]
-then
-    set -x
+usage() {
+  cat << EOF
+usage: bash remove-webhook.sh [--insecure-skip-tls-verify]
+
+This script adjusts the rancher-webhook chart release in all clusters managed by Rancher (excluding the local cluster).
+Depending on the version of Rancher, it either deletes the downstream webhook release, adjusts its version and restarts, or does nothing.
+Requires kubectl and helm to be installed and available on \$PATH.
+
+RANCHER_URL without a trailing slash must be set with the server URL of Rancher.
+RANCHER_TOKEN must be set with an admin token generated with no scope.
+To ignore TLS verification, set --insecure-skip-tls-verify.
+
+Users also need to ensure they have the rancher-charts repo in the local Helm index.
+EOF
+}
+
+if [ "$1" == "-h" ]; then
+  usage
+  exit 0
 fi
 
-usage() {
-    echo "./remove-webhook.sh [--insecure-skip-tls-verify]"
-    echo "Remove the webhook chart in all clusters managed by rancher (excluding the local cluster)"
-    echo "Requires kubectl and helm to be installed and available on \$PATH"
-    echo "--insecure-skip-tls-verify can be set to configure the script to ignore tls verification"
-    echo "RANCHER_TOKEN must be set with an admin token generated with no scope"
-    echo "RANCHER_URL must be set with the url of rancher (no trailing /) - should be the server URL"
+delete_webhook() {
+  cluster="$1"
+  current_chart=$(helm list -n cattle-system -l name=rancher-webhook | tail -1 | cut -f 6)
+  echo "Deleting $current_chart from cluster $cluster."
+  helm uninstall rancher-webhook -n cattle-system
+}
+
+replace_webhook() {
+  cluster="$1"
+  new_version="$2"
+
+  echo "Updating the agent to make it remember the min version $new_version of rancher-webhook, so that it can deploy it when needed in the future in cluster $cluster."
+  kubectl set env -n cattle-system deployment/cattle-cluster-agent CATTLE_RANCHER_WEBHOOK_MIN_VERSION="$new_version"
+
+  helm get values -n cattle-system rancher-webhook > current_values.yaml
+  delete_webhook "$cluster"
+  echo "Re-installing rancher-webhook to use $new_version in cluster $cluster."
+  helm upgrade --install rancher-webhook rancher-charts/rancher-webhook -n cattle-system --version "$new_version" --values current_values.yaml
+  rm -f current_values.yaml
+}
+
+adjust_webhook() {
```

Comment:
As far as I can tell, this does not cover the use case of 2.6 versions before 2.6.13. We want to cover these cases in case of rollback from 2.6.13+ to <2.6.13 (e.x. 2.6.8), in these cases we want to delete the webhook.

Reply:
Good catch!